### ISSUE TITLE
Adding own contexts keys makes "sentry" contexts key not added

### ISSUE THREAD

**Issue by MaxLeb** [getsentry/sentry-native#680]
**Description**

When you add a key "contexts" in a "manual" events, the usually added keys by sentry are not added.

So if you want to add extra contexts to a manual event, then you will loose all OS informations and such that are usually automatically added by Sentry.

Based on your example for manual events, the following code will have OS information

```
sentry_value_t event = sentry_value_new_event();
sentry_value_set_by_key(event, "message", sentry_value_new_string("Hello!"));

sentry_capture_event(event);
```

while the following code will not have it

```
sentry_value_t screen = sentry_value_new_object();
sentry_value_set_by_key(screen, "width", sentry_value_new_int32(1920));
sentry_value_set_by_key(screen, "height", sentry_value_new_int32(1080));

sentry_value_t contexts = sentry_value_new_object();
sentry_value_set_by_key(contexts, "screen_size", screen);

sentry_value_set_by_key(event, "contexts", contexts);
sentry_capture_event(event);
```

**When does the problem happen**

- [ ] During build
- [x] During run-time
- [ ] When capturing a hard crash

**Environment**

This is not env dependant.
Using SDK 0.4.13

**Steps To Reproduce**

Add this snippet to any application that communicates with Sentry:

```
sentry_value_t screen = sentry_value_new_object();
sentry_value_set_by_key(screen, "width", sentry_value_new_int32(1920));
sentry_value_set_by_key(screen, "height", sentry_value_new_int32(1080));

sentry_value_t contexts = sentry_value_new_object();
sentry_value_set_by_key(contexts, "screen_size", screen);

sentry_value_set_by_key(event, "contexts", contexts);
sentry_capture_event(event);
```

In Sentry web interface, you will not see the OS information.

Workaround: use `sentry_set_context` to add your extra informations then right after the `sentry_capture_event` use `sentry_remove_context` to ensure that your extra contexts are not kept for other events.

**Comment by Swatinem** [comment]
Can you update to the latest SDK release? We actually implemented context merging in `0.4.14` (though it seems I missed that from the changelog).

**Comment by MaxLeb** [comment]
Seems fixed with 0.4.15 :)
